src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -1283,48 +1283,26 @@ public void postBlUsersPasswordResetReset(String userAgent, String xForwardedFor
     Optional.ofNullable(xForwardedFor)
       .ifPresent(header -> requestHeaders.put(X_FORWARDED_FOR_HEADER, header));
 
-    String tokenSub = getPayloadFromToken(okapiHeaders);
-    if (!tokenSub.startsWith(UNDEFINED_USER_NAME)) {
-      UnprocessableEntityMessage message = new UnprocessableEntityMessage(LINK_INVALID_STATUS_CODE,
-        "Invalid token.");
-      asyncResultHandler.handle(Future.succeededFuture(
-        ExceptionHelper.handleException(new UnprocessableEntityException(Collections.singletonList(message)))));
-      return;
-    }
-
-    String passwordResetActionId = tokenSub.substring(UNDEFINED_USER_NAME.length());
-    passwordResetLinkService.resetPassword(passwordResetActionId, request.getString("newPassword"),
-      requestHeaders).setHandler(res -> {
-        if (res.succeeded()) {
-          asyncResultHandler.handle(Future.succeededFuture(PostBlUsersPasswordResetResetResponse.respond204()));
-        } else {
-          asyncResultHandler.handle(Future.succeededFuture(ExceptionHelper.handleException(res.cause())));
-        }
-    });
-
+    passwordResetLinkService.resetPassword(request.getString("newPassword"), requestHeaders)
+      .map(PostBlUsersPasswordResetResetResponse.respond204())
+      .map(javax.ws.rs.core.Response.class::cast)
+      .otherwise(ExceptionHelper::handleException)
+      .setHandler(asyncResultHandler);
   }
 
   @Override
   public void postBlUsersPasswordResetValidate(Map<String, String> okapiHeaders,
                                                Handler<AsyncResult<javax.ws.rs.core.Response>> asyncResultHandler,
                                                Context vertxContext) {
     OkapiConnectionParams connectionParams = new OkapiConnectionParams(okapiHeaders.get(BLUsersAPI.OKAPI_URL_HEADER),
```

Comment:
I suppose this code was here before, but it can be replaced with 
```OkapiConnectionParams connectionParams = new OkapiConnectionParams(okapiHeaders);```

Reply:
Made changes to the OkapiConnectionParams